### PR TITLE
VACMS-16348: Service location ux enhancements

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #16348 

### When "Use default text" is checked
Screenshot 2024-02-01 at 5 23 34 PM

### When "Customize text" is checked
Screenshot 2024-02-01 at 5 23 19 PM

### When "Remove text" is checked
Screenshot 2024-02-01 at 5 21 55 PM

### What needs to be checked to prove this works?
- [x] Assign QA Content Publisher the following roles:
  - [x] Content creator - VAMC
  - [x] Content creator - VBA
  - [x] Content publisher
- [x] Assign QA Content Publisher to the following Sections:
  - [x] VA Southeast Louisiana health care
  - [x] Des Moines VA Regional Benefit Office

The following spreadsheet identifies the testing scenarios:
https://docs.google.com/spreadsheets/d/1JxdTIbIibkY3UQ3DIix9hr7XcP4CaueL_1sYDHJIKmg/edit#gid=221281300

For each row, 
1. Mark or fill in the field(s) along with any required Service location or node fields.
2. Confirm the appearance or disappearance of the Default text.
3. Confirm the appearance or disappearance of the Custom text field.
4. Save the node.
5. Edit the node to review the Custom text field result.

#### Users

- For Service region, log in as an admin.
- For all other content types, log in as QA Content Publisher.

Note: Each test depends on the presence of a second service location. You may need to add the other.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: docroot/modules/custom/va_gov_facilities/va_gov_facilities.info.yml [hunk 1]
```diff
@@ -3,3 +3,5 @@ type: module
 description: 'Tools to improve user experience for VA Facilities.'
 core_version_requirement: ^9 || ^10
 package: 'Custom'
+dependencies:
+  - va_gov_backend
```
Comment (Becapa) [inline docroot/modules/custom/va_gov_facilities/va_gov_facilities.info.yml:7]: The code in the backend module that this was previously dependent upon (even though it wasn't listed as a dependency then) was moved out and into the facilities module with these changes. Does this need to be listed as a dependency now?

**Review by Becapa — APPROVED** [review]
I think this looks good. I had one comment that I added, but it's definitely not blocking. I also noticed most of the functions here have an if statement that wraps the entire function. I wonder if at some point we could talk about the guard clause coding pattern and whether it's something we'd like to use moving forward or not.

File: docroot/modules/custom/va_gov_facilities/va_gov_facilities.info.yml (hunk 1 above)
Comment (omahane) [reply docroot/modules/custom/va_gov_facilities/va_gov_facilities.info.yml:7]: Yes, as I found a method in va_gov_backend code that is being called in this module. I should have explained that.

**Comment by omahane** [comment]
> I think this looks good. I had one comment that I added, but it's definitely not blocking. I also noticed most of the functions here have an if statement that wraps the entire function. I wonder if at some point we could talk about the guard clause coding pattern and whether it's something we'd like to use moving forward or not.

Yes, that's a good idea. I love it and will implement it going forward, unless you feel like it needs it here.

**Comment by Becapa** [comment]
> > I think this looks good. I had one comment that I added, but it's definitely not blocking. I also noticed most of the functions here have an if statement that wraps the entire function. I wonder if at some point we could talk about the guard clause coding pattern and whether it's something we'd like to use moving forward or not.
> 
> Yes, that's a good idea. I love it and will implement it going forward, unless you feel like it needs it here.

No, I don't think it needs it here. It's just more of a preference for readability and none of the code here is so nested that it's unreadable so I think it's good.

**Comment by davidmpickett** [comment]
The ticket says "Deprecated fields are hidden" but they are still showing in the Tugboat. I feel like maybe you and Steve had a conversation about how hiding the fields should wait until after the migration? Will that be a separate PR?

1C5C7E11-0BC9-490A-BD79-F2DF32BC217D_1_201_a

**Comment by omahane** [comment]
> The ticket says "Deprecated fields are hidden" but they are still showing in the Tugboat. I feel like maybe you and Steve had a conversation about how hiding the fields should wait until after the migration? Will that be a separate PR?
> 
Yes, that's going to be separate.

**Comment by swirtSJW** [comment]
Nice work @omahane.   I am merging this into integration now.  So we will need a followup PR to hide the fields.

**Pull request merged**